tests/garage/tf/policies/test_gaussian_mlp_policy_with_model.py

```diff
@@ -0,0 +1,181 @@
+        entropy1 = entropy_func(obs)  # result
+
+        # * New way *
+        # equvaient to garage.tf.distribution.entropy_sym(dist)
+        entropy_sym2 = policy3.model.networks['default'].distribution.entropy()
+        entropy2 = self.sess.run(  # result
+            entropy_sym2,
+            feed_dict={policy3.model.networks['default'].input: obs})
+
+        assert entropy1 == entropy2
+
+        ############################################################
+        # likelihood_ratio_sym
+        # * Existing way *
+        likelihood_ratio_sym1 = policy1.distribution.likelihood_ratio_sym(
+            obs_ph, dist1_sym, dist2_sym, name='li_ratio_sym1')
+        likelihood_ratio_func = tensor_utils.compile_function(
+            [obs_ph], likelihood_ratio_sym1)
+        likelihood_ratio1 = likelihood_ratio_func(obs)
+
+        # * New way *
+        # tf.distributions seems doesn't have this available
+        # maybe do it ourselves
+        with tf.name_scope('li_ratio_sym2'):
+            log_prob_diff = policy4.model.networks[
+                'default'].distribution.log_prob(
+                    obs_ph, name='log_prob_obs') - policy3.model.networks[
+                        'default'].distribution.log_prob(obs_ph)
+
+            likelihood_ratio_sym2 = tf.exp(log_prob_diff)
+            likelihood_ratio2 = self.sess.run(  # result
+                likelihood_ratio_sym2,
+                feed_dict={
+                    policy3.model.networks['default'].input: obs,
+                    policy4.model.networks['default'].input: obs,
+                    obs_ph: obs
+                })
+
+        assert likelihood_ratio1 == likelihood_ratio2
+
+    def test_guassian_mlp_policy_pickle(self):
+        box_env = TfEnv(DummyBoxEnv())
+        data = np.ones((3, 1))
+        with tf.Session(graph=tf.Graph()) as sess:
+            policy = GaussianMLPPolicyWithModel2(
+                env_spec=box_env, init_std=1.1)
+            # model is built in GaussianMLPPolicyWithModel2.__init__
+            outputs = sess.run(
+                policy.model.networks['default'].sample,
+                feed_dict={policy.model.networks['default'].input: data})
+            p = pickle.dumps(policy)
+
+        with tf.Session(graph=tf.Graph()) as sess:
+            policy_pickled = pickle.loads(p)
+            # After pickle, we need to build the model
+            # e.g. by policy.dist_info_sym
+            input_ph = box_env.observation_space.new_tensor_variable(
+                extra_dims=1, name='input_ph')
+            policy_pickled.dist_info_sym(
+                input_ph, tfp.distributions.MultivariateNormalDiag)
+
+            outputs2 = sess.run(
+                policy_pickled.model.networks['default'].sample,
+                feed_dict={
+                    policy_pickled.model.networks['default'].input: data
+                })
+
+        assert np.array_equal(outputs, outputs2)
```

Comment:
Are we abandoning `tensor_utils.compile_function()`? In this test, all the new ways use `sess.run`.

Reply:
I am not sure. That was discussed in another PR. Using `sess.run` is just my personal preference.

Reply:
i have no preference. do whatever seems best for now and we can fix this style in a different PR